test/e2e/tests/risingwave/meta.sh

```diff
@@ -14,28 +14,58 @@
 
 ${__E2E_SOURCE_TESTS_RISINGWAVE_META_SH__:=false} && return 0 || __E2E_SOURCE_TESTS_RISINGWAVE_META_SH__=true
 
-function risingwave::utils::kill_the_meta_leader_pod() {
-  local meta_names
-  meta_names="$(k8s::kubectl::get pod -l risingwave/component=meta -o=jsonpath='{.items..metadata.name}')"
-  if [ -z "$meta_names" ]; then
-    logging::error "failed to retrieve meta nodes"
+function risingwave::utils::delete_leader_lease() {
+  local meta_leaders_pod_IPs
+  meta_leaders_pod_IPs="$(k8s::kubectl::get pod -l risingwave/component=meta -l risingwave/meta-role=leader -o=jsonpath='{.items..status.podIP}')"
+
+  if [ -z "$meta_leaders_pod_IPs" ]; then
+    logging::error "No meta leader node found"
+    return 1
+  fi
+
+  if [ "$(echo "$meta_leaders_pod_IPs" | wc -l)" -gt 1 ]; then
+    logging::error "More than one meta leader node found"
     return 1
   fi
 
-  local leader=""
-  for p in $meta_names; do
-    local is_leader
-    is_leader=$(k8s::kubectl logs "$p" | grep "Defining leader services")
-    if [ "$is_leader" != "" ]; then
-      if [ "$leader" != "" ]; then
-        logging::error "Split brain detected! $p and $leader are leaders!"
-        return 1
-      fi
-      leader="$p"
+  # Iterate over the etcd election kv pairs. Delete leader lease if found, else abort the test
+  local del_lease=false
+  kubectl -n rwc-2-mytenant run del-leader-lease --image='anldisr/etcdctl' --restart='Never' --command=true -- etcdctl --endpoints=risingwave-etcd-mytenant:2379 get _ --prefix=true --write-out="json"
```

Comment:
As always thank you for your guidance 😁 

I totally agree with you, using `exec` instead of spewing a new pod is definitely better. 

**Regarding performance:** I am not sure how we could use `lease list` in this case. That command returns 

```json 
{
  "cluster_id": 6604019301136558000,
  "member_id": 1109445486625178000,
  "revision": 13954,
  "raft_term": 4,
  "leases": [
    {
      "id": 4734838729492364000
    },
    {
      "id": 4734838729492364000
    },
    {
      "id": 4734838729492364000
    }
  ]
}
```

We receive the leases, but we do not know which keys are attached to these leases. I do not see how this benefits us here. Please let me know if I miss something. 

I will use `etcdctl get __meta_election_ --prefix=true` from now on. Maybe that is a bit more performant, since we narrow down the search a bit more

Reply:
Never mind. I believe `etcdctl get __meta_election_ --prefix=true` already solves the problem. I didn't know more about the `lease list` than you.